AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -3468,6 +3440,27 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   return [self layoutThatFits:constrainedSize parentSize:constrainedSize.max];
 }
 
+- (void)setPreferredFrameSize:(CGSize)preferredFrameSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+
+  // Deprecated preferredFrameSize just calls through to set width and height
+  self.width = ASDimensionMake(preferredFrameSize.width);
+  self.height = ASDimensionMake(preferredFrameSize.height);
+  [self invalidateCalculatedLayout];
```

Comment:
Can we remove this invalidation? Shouldn't `width` and `height` perform an equality check and invalidate if needed?